src/modules/FacilitesModule/components/FacilitesList/FacilitesList.tsx

```diff
@@ -1,8 +1,435 @@
+    } catch (error: any) {
+      toast.error(error.response.message || "erroe");
+    }
+  }
+  const handleMenuClick = (
+    event: React.MouseEvent<HTMLElement>,
+    rowId: string
+  ) => {
+    setAnchorEl(event.currentTarget);
+    setSelectedRow(rowId);
+  };
+
+  const handleMenuClose = () => {
+    setAnchorEl(null);
+    setSelectedRow(null);
+  };
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  useEffect(() => {
+    getFacilites();
+  }, []);
+
   return (
-    <div>FacilitesList</div>
-  )
-}
+    <>
+      <Modal
+        open={deleteopen}
+        aria-labelledby="transition-modal-title"
+        aria-describedby="transition-modal-description"
+        closeAfterTransition
+        slots={{ backdrop: Backdrop }}
+        slotProps={{
+          backdrop: {
+            timeout: 600,
+          },
+        }}
+      >
+        <Box
+          sx={{
+            margin: "auto",
+            textAlign: "center",
+            position: "absolute",
+            top: "50%",
+            left: "50%",
+            transform: "translate(-50%, -50%)",
+            bgcolor: "background.paper",
+            boxShadow: 3,
+            p: 6,
+            maxWidth: 500,
+          }}
+        >
+          <DeleteData
+            title="Facility"
+            item="Facility"
+            closing={handleCloseDelete}
+          />
+          <Stack mt={5}>
+            <Button variant="outlined" color="error" onClick={DeleteFacility}>
+              Delete
+            </Button>
+          </Stack>
+        </Box>
+      </Modal>
+      {/* //Model */}
+
+      <BootstrapDialog
+        onClose={handleClose}
+        aria-labelledby="customized-dialog-title"
+        open={open}
+        PaperProps={{
+          component: "form",
+          onSubmit: handleSubmit(onSubmit),
+        }}
+      >
+        <DialogTitle
+          sx={{ m: 0, p: 2, fontWeight: "bold" }}
+          id="customized-dialog-title"
+        >
+          {modalState === "update" ? "Updata" : "Add"} Facilites
+        </DialogTitle>
+        <IconButton
+          aria-label="close"
+          onClick={handleClose}
+          sx={{
+            position: "absolute",
+            right: 8,
+            top: 8,
+            color: (theme) => theme.palette.grey[500],
+          }}
+        >
+          <CloseIcon />
+        </IconButton>
+        <DialogContent dividers>
+          <TextField
+            fullWidth
+            margin="normal"
+            required
+            placeholder="Name"
+            sx={{ marginBottom: 7, background: "#F5F6F8", marginTop: 7 }}
+            {...register("name", { required: "Name Is Requird" })}
+          />
+        </DialogContent>
+        <DialogActions>
+          {modalState === "update" ? (
```

Comment:
We can make the condition around the text instead, since it is the only thing that changes in the button.